### ISSUE TITLE
Redirection is not working after click to Accept

### ISSUE THREAD

**Issue by Eldenroot** [kawaii/mybb-amnesia#4]
Maybe would be nice to redirect user to the main page (or configurable URL)

**Comment by dvz** [comment]
Take a look at the value for the `url_before_redirect` form field (basing on `$_SERVER['HTTP_REFERER']`) on the _Accept_ page, respected if this condition is met:
https://github.com/kawaii/mybb-amnesia/blob/15e2115/inc/plugins/amnesia/hooks_frontend.php#L307

**Comment by Eldenroot** [comment]
@dvz - ok, there is an issue if the user does not have any "previous pages" visited, so the redirect returns back to GDPR page over and over. Maybe would be nice in that cases to redirect this user to forum homepage.

**Comment by dvz** [comment]
It should redirect to _Board URL_, unless it's not not set properly; another reason for a loop is an update date set in the future (in this case the _Accept_ button would keep appearing).

**Comment by IntoxNitram** [comment]
Yeah I think it's not working properly. I have a date set in the past and if the user logs in and has to accept it just accepts but redirects to the policy page again (no accept button).

**Comment by andrewjs18** [comment]
I personally cannot duplicate this...it seems to work as expected for me on a test board with no other plugins installed.

that being said, I did notice that when you turn off friendly redirects and you accept the new policy, it does not remove the message until you click on another part of the site to get it to load a new page.  this could probably be fixed, I assume, by forcing the page to go to the board url after a user clicks 'accept' if they have friendly redirects turned off.

**Comment by IntoxNitram** [comment]
It’ll probably be the friendly redirects bit then, as I have that turned off on my forum.

**Comment by andrewjs18** [comment]
weird..the message did go away after clicking 'accept' this time with friendly redirects turned off.  earlier it wasn't.  

this is what it'd look like earlier after accepting the policy with the redirects turned off:

![2018-05-28 16_55_36](https://user-images.githubusercontent.com/2748980/40629276-111aff54-6298-11e8-9f82-ec7183a09e93.png)

**Comment by IntoxNitram** [comment]
Yep it seems inconsistent I’ve just tried it again on another account on my live forum and it redirected correctly back to the index this time.
